tools/deb/BUILD

```diff
@@ -0,0 +1,67 @@
+load("@bazel_tools//tools/build_defs/pkg:pkg.bzl", "pkg_tar", "pkg_deb")
+
+pkg_tar(
+    name = "agent-bin",
+    srcs = [
+        "//cmd/pilot-agent:pilot-agent",
+    ],
+    mode = "0755",
+    package_dir = "/usr/local/bin",
+)
+
+pkg_tar(
+    name = "discovery-bin",
+    srcs = [
+        "//cmd/pilot-discovery:pilot-discovery",
+    ],
+    mode = "0755",
+    package_dir = "/usr/local/bin",
+)
+
+pkg_tar(
+    name = "debian-data-agent",
+    extension = "tar.gz",
+    deps = [
+        ":agent-bin",
+    ],
+)
+
+pkg_tar(
+    name = "discovery-systemd",
+    srcs = ["istio-discovery.service"],
+    mode = "644",
+    package_dir = "/lib/systemd/system",
+)
+
+pkg_tar(
+    name = "debian-data-discovery",
+    extension = "tar.gz",
+    deps = [
+        ":discovery-bin",
+        ":discovery-systemd",
+    ],
+)
+
+pkg_deb(
+    name = "istio-agent",
+    architecture = "amd64",
+    built_using = "bazel",
+    data = ":debian-data-agent",
+    description_file = "description-agent",
+    homepage = "http://istio.io",
+    maintainer = "The Istio Authors <istio-dev@googlegroups.com>",
+    package = "istio-agent",
+    version = "0.2.1",
```

Comment:
there seems to be many cases where packages don't build because BUILD is too generic, I think BUILD.bazel is a lot clearer / specific (this is a bazel build file, it can't be used by anything else than bazel (afaik)) - but we can do a mass rename later

ps: BUILD.bazel is what gazelle generates